### PR TITLE
Grid toolbar reworked with flex

### PR DESCRIPTION
**Pull request by GeorgeGindev**
Targets #2037

### CODE REVIEW

File: packages/default/scss/grid/_layout.scss [hunk 1]
```diff
@@ -364,9 +364,9 @@
             }
 
             .k-grid-toolbar {
-
-                .k-button + .k-button {
-                    margin: 0 $grid-toolbar-spacing 0 0;
+                padding: $grid-toolbar-padding-y $grid-toolbar-padding-x 0 0;
+                > * {
```
Comment (joneff) [inline packages/default/scss/grid/_layout.scss:368]: If we are using flex, we can use `gap`

File: packages/default/scss/grid/_layout.scss [hunk 2]
```diff
@@ -376,7 +376,8 @@
             }
 
             .k-grid-search {
-                float: left;
+                margin-left: $grid-toolbar-spacing;
```
Comment (joneff) [inline packages/default/scss/grid/_layout.scss:379]: no need here for those.

see if you can manually place a spacer

File: packages/default/scss/grid/_layout.scss [hunk 3]
```diff
@@ -413,7 +414,7 @@
 
         .k-grid-search {
             width: $grid-search-width;
-            float: right;
+            margin-left: auto;
```
Comment (joneff) [inline packages/default/scss/grid/_layout.scss:417]: probably this will also not be needed

File: tests/visual/grid-toolbar.html [hunk 4]
```diff
@@ -60,9 +60,9 @@
                 <div class="k-header k-grid-toolbar">
                     <button class="k-button">Button</button>
                     <span class="k-button-group">
-                        <button class="k-button-icon k-button k-group-start"><span class="k-icon k-i-bold"></span></button>
+                        <button class="k-button-icon k-button k-group-end"><span class="k-icon k-i-bold"></span></button>
```
Comment (joneff) [inline tests/visual/grid-toolbar.html:63]: Why? The first button is the start button. If for some reason the buttons do not look correctly, it's not issue of the class names, rather, the styles. You can look at the button-group styles for a fix.

**Review by joneff — CHANGES_REQUESTED** [review]

**Comment by joneff** [comment]
Quick question: we already have the styles for toolbar. What if we add the class for toolbar to the grid toolbar:

```html<!-- current -->
<div class="k-header k-grid-toolbar">
    ...
</div>


<!-- suggestion -->
<div class="k-toolbar k-grid-toolbar">
    ...
</div>
```

Looking at the styles for toolbar, that will make the items wrap on small resolutions with the appropriate spacing. Being a styled toolbar, we can also use spacers. Finally, that will also clean up some variables, as we won't need them [when we have the classnames for styling toolbar size and color wise].

@VicTachev @tapopov @Xizario @Raisolution @inikolova thoughts?

**Comment by joneff** [comment]
@silviyaboteva @inikolova we'll be moving the styles to dev!

**Review by joneff — APPROVED** [review]

**Review by PreslavKozovski — APPROVED** [review]

**Review by inikolova — CHANGES_REQUESTED** [review]
Add **flex-shrink: 0** to **.k-grid-toolbar** otherwise toolbar squashes when the Grid is scrollable and small height is set

**Review by inikolova — APPROVED** [review]

**Comment by ivanchev** [comment]
@joneff can myself and @jivanova get a mention when such changes are required in the future, since right now the Blazor Grid is broken, and we had no idea that this is going to happen.

**Comment by joneff** [comment]
An oversight on my part. I thought I mentioned all managers in the previous comment.

**Pull request merged**